crates/metrics/metrics-derive/tests/trybuild.rs

```diff
@@ -0,0 +1,5 @@
+#[test]
+fn compile_test() {
+    let t = trybuild::TestCases::new();
```

Comment:
also really neat - TIL

Reply:
> First, if a test case is being run as compile_fail but a corresponding *.stderr file does not exist, the test runner will save the actual compiler output with the right filename into a directory called wip within the directory containing Cargo.toml. So you can update these files by deleting them, running cargo test, and moving all the files from wip into your testcase directory.

> Alternatively, run cargo test with the environment variable TRYBUILD=overwrite to skip the wip directory and write all compiler output directly in place. You'll want to check git diff afterward to be sure the compiler's output is what you had in mind.

https://github.com/dtolnay/trybuild#workflow - really sick work @dtolnay, thanks for the crate

Reply:
i've spent a couple of hours battling `compiletest-rs` and rustc lib linkage before finding this gem. indeed, great work, really easy to use